ui/plugin-system/src/runtime/DataQueriesProvider/DataQueriesProvider.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { createContext, useCallback, useContext, useMemo } from 'react';
+import { Definition, TimeSeriesQueryDefinition, UnknownSpec } from '@perses-dev/core';
+import { useTimeSeriesQueries } from '../time-series-queries';
+import { DataQueriesProviderProps, QueryData, UseDataQueriesResult } from './model';
+
+export function useDataQueries(): UseDataQueriesResult {
```

Comment:
Nit: I wonder if we should rename this to be more clear that it returns the data?
```
useQueriesData()
useQueryDataResults()
```